core/src/main/java/com/linecorp/armeria/server/encoding/HttpEncodedResponse.java

```diff
@@ -76,8 +76,8 @@ protected HttpObject filter(HttpObject obj) {
             final ResponseHeaders headers = (ResponseHeaders) obj;
 
             // Skip informational headers.
-            final String status = headers.get(HttpHeaderNames.STATUS);
-            if (ArmeriaHttpUtil.isInformational(status)) {
+            final HttpStatus status = headers.status();
+            if (status.isInformational() || status.isContentAlwaysEmpty()) {
```

Comment:
Can we move this check to `shouldEncodeResponse`?

Reply:
Ah thanks. It should be.